### ISSUE TITLE
Extending canary testing into A/B testing

### ISSUE THREAD

**Issue by klee989** [kserve/kserve#335]
/kind feature

**Describe the solution you'd like**
Could be useful to extend the existing implementation of canary testing into A/B testing: routing between versions, collecting an aggregate metric for each version, and computing a test of significance between versions' metrics. To discuss in Wednesday's meeting?

**Anything else you would like to add:**

**Comment by yuzisun** [comment]
cc @ellis-bigelow @cliveseldon @rakelkar @animeshsingh

**Comment by yuzisun** [comment]
@cliveseldon Is this fall under MLGraph ?

**Comment by cliveseldon** [comment]
Yes I think this would fall under MLGraph proposal I did to add A/B tests, multi-armed bandits and ensembling. I did an initial spec in the Seldon repo: https://github.com/SeldonIO/mlgraph but these concepts will need to be discussed by the kfserving group as to what makes sense.

**Comment by rakelkar** [comment]
AB/flighting seems to have the following differences when compared to rollout:
1. May fork a little bit of traffic to many services that participate in an overall experiment
2. Not roll-forward, may just be a timebound fork to get data
3. Potentially finer traffic distribution granularity - may be determined by an external control tower like system on the basis of a set of variants e.g. 1% of mobile traffic

**Comment by animeshsingh** [comment]
This would need a monitoring solution to be in place, and maybe a good segway into that discussion.

**Comment by nurcan-sonmez** [comment]
Hello hello, are there any plans to roll this out with the next version? Also, any plans to include multi armed bandit in addition to A/B testing?

**Comment by spolti** [comment]
Does it is still an issue?
It seems that split the traffic is possible: https://kserve.github.io/website/0.13/modelserving/v1beta1/rollout/canary/
May we close it? cc @yuzisun @sivanantha321
